crates/goose/src/execution/manager.rs

```diff
@@ -91,43 +91,40 @@ impl AgentManager {
         session_id: String,
         mode: SessionExecutionMode,
     ) -> Result<Arc<Agent>> {
-        let agent = {
+        let (agent, is_new) = {
             let mut sessions = self.sessions.write().await;
-            if let Some(agent) = sessions.get(&session_id) {
-                debug!("Found existing agent for session {}", session_id);
-                return Ok(Arc::clone(agent));
+            if let Some(existing) = sessions.get(&session_id) {
+                (Arc::clone(existing), false)
+            } else {
+                let agent = Arc::new(Agent::new());
+                sessions.put(session_id.clone(), agent.clone());
+                (agent, true)
             }
-
-            info!(
-                "Creating new agent for session {} with mode {}",
-                session_id, mode
-            );
-            let agent = Arc::new(Agent::new());
-            sessions.put(session_id.clone(), Arc::clone(&agent));
-            agent
         };
 
-        match &mode {
-            SessionExecutionMode::Interactive | SessionExecutionMode::Background => {
-                debug!("Setting scheduler on agent for session {}", session_id);
-                agent.set_scheduler(Arc::clone(&self.scheduler)).await;
+        if is_new {
+            match &mode {
+                SessionExecutionMode::Interactive | SessionExecutionMode::Background => {
+                    debug!("Setting scheduler on agent for session {}", session_id);
+                    agent.set_scheduler(Arc::clone(&self.scheduler)).await;
```

Comment:
I know this is existing code but why do we `agent.set_scheduler` for an interactive session?

Reply:
it's not a great reason, but goose can create schedules for itself and in order to that, the agent needs to have access to the scheduling service